### ISSUE TITLE
conversion to SBU format is broken

### ISSUE THREAD

**Issue by kaczmarj** [SBU-BMI/wsinfer#186]
At the moment, this error is raised, and I assume there will be other errors as well if that one is fixed:

```python
  File "/xxxxxxx/wsinfer/cli/convert_csv_to_sbubmi.py", line 342, in tosbu
    class_names: list[str] = run_metadata["model_weights"]["class_names"]
KeyError: 'model_weights'
```

**Comment by kaczmarj** [comment]
@swaradgat19 - could you please look into this?

**Comment by kaczmarj** [comment]
the conversion script also needs to be updated to glob multiple image file types, not just SVS files.

**Comment by swaradgat19** [comment]
Sure. I'll look into it

**Comment by swaradgat19** [comment]
@kaczmarj Could you tell me a little about `execution-id`, `study-id` and `make-color-text` fields in the `tosbu` command?

**Comment by kaczmarj** [comment]
>kaczmarj Could you tell me a little about execution-id, study-id and make-color-text fields in the tosbu command?

of course, i will be honest, i am a bit fuzzy about this format. i learned about the format by reviewing the structure of several existing files.

- `execution-id` - a unique ID for the run. I believe this is a UUID (which you can create in python using the `uuid` module in the standard library).
- `study-id` - this is a string that identifies the group of people or samples that were analyzed. this is an arbitrary string, like `TCGA-BRCA`).
- `make-color-text` - the format includes two files. one of the files is a text file that contains color information of each patch. i am not sure if this color file was always required, so i left it as an option. the creation of this color text file can take a considerable amount of time because the patches need to be loaded from the whole slide image.

let me know if you have any other questions.

**Comment by swaradgat19** [comment]
Got it. So if I want to run this `tosbu` command, what should I put as the `execution-id`?

**Comment by kaczmarj** [comment]
for now, use a UUID. i show how to make one in python below:

```python
import uuid
execution_id = str(uuid.uuid4())
```

**Comment by kaczmarj** [comment]
closed in favor of #113
